File: 30 days of code/Day 16 - Exceptions (String to integer)/Exceptions (String to Integer).java
/**
 *              EXCEPTIONS (STRING TO INTEGER) | JAVA
 * Domain       30 DAYS OF CODE
 * Sub domain   DAY 16
 */

import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) throws FileNotFoundException {
        File file = new File("input.txt");
        Scanner scan = new Scanner(file);
        String input = scan.next();
        scan.close();

        try {
            int n = Integer.parseInt(input);
            System.out.println(n);
        } catch (NumberFormatException nfe) {
            System.out.println("Bad String");
        }
    }
}

File: 30 days of code/Day 16 - Exceptions (String to integer)/Exceptions (String to Integer).cs
/**
 *              EXCEPTIONS (STRING TO INTEGER) | C#
 * Domain       30 DAYS OF CODE
 * Sub domain   DAY 16
 */

using System;
using System.IO;

class Program
{
    static void Main(string[] args)
    {
        var reader = new InputReader(false);
        string input = reader.NextString;
        reader.Close();

        try
        {
            int n = int.Parse(input);
            Console.WriteLine(n);
        }
        catch (FormatException)
        {
            Console.WriteLine("Bad String");
        }
    }
}

class InputReader
{
    private TextReader input;

    public InputReader(bool fromFile)
    {
        input = fromFile ? new StreamReader("input.txt") : Console.In;
    }

    public string NextLine => input.ReadLine();

    public string NextString => NextLine;

    public void Close() => input.Close();
}
